### PR TITLE
ESC-1411 - removed unneeded optional email form. Re-routed journeys a…

### PR DESCRIPTION
**Pull request by WitchellJack**
…nd deleted now redundant tests related to optional email field.

### CODE REVIEW

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BecomeLeadController.scala [hunk 1]
```diff
@@ -93,9 +93,10 @@ class BecomeLeadController @Inject() (
     logger.info("BecomeLeadController.postConfirmEmail")
     handleConfirmEmailPost[BecomeLeadJourney](
       previous = routes.BecomeLeadController.getConfirmEmail.url,
```
Comment (habeeb01) [inline app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BecomeLeadController.scala:95]: should previous and current have the same value? looks odd to me (i might be missing something)

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala [hunk 2]
```diff
@@ -207,6 +207,7 @@ class UndertakingController @Inject() (
       val backLink = status match {
```
Comment (habeeb01) [inline app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala:207]: create a function for this so that its not repeated.

**Comment by habeeb01** [comment]
I can't see tests to make sure that: -
- user is going to the right url when they select no
- the back link is correct when the user is on the input email address page

File: app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BecomeLeadController.scala (hunk 1 above)
Comment (WitchellJack) [reply app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BecomeLeadController.scala:95]: Yes, good spot and I think this is actually the location of that existing bug related to the back link! have fixed this so you should see that reflected in latest commit 👍🏻

File: conf/messages [hunk 3]
```diff
@@ -170,6 +170,7 @@ email.error.required = Enter the email address you want to use to receive notifi
 email.error.email = Enter a valid email address
 email.error.maxLength = The email address for your undertaking must be 254 characters or less
 using-stored-email.error.required = Select yes if this is the email address you want to use to receive notifications
+using-stored-email.error.using-stored-email.required = Select yes if this is the email address you want to use to receive notifications
```
Comment (habeeb01) [inline conf/messages:173]: why is this needed?

**Review by habeeb01 — DISMISSED** [review]

**Review by feyyazrc — APPROVED** [review]
